### PR TITLE
[#12293] Admin logs page: Extension of query start/end time is not reflected in search range

### LINKED ISSUE #12293: Admin logs page: Extension of query start/end time is not reflected in search range
Currently, clicking the extend start/end time buttons causes the query time range to be changed, but the change is not reflected in the user input. In the below example, the start time was extended by 20 minutes to 13:42, but the user input remains at 14:02.
Screenshot 2023-03-28 at 17 06 25

Suggestion: After clicking the extend start/end time buttons, the user input date range should also be changed accordingly.

### PR DESCRIPTION
**Pull request by gordonscottkolb**
Fixes #12293

### CODE REVIEW

**Comment by kevin9foong** [comment]
hi @gordonscottkolb, thanks for the PR! I will be reviewing this PR shortly.

File: src/web/services/datetime.service.ts [hunk 1]
```diff
@@ -52,6 +52,33 @@ export class DateTimeService {
...
+    convertDateToDateFormatAndTimeFormat(date: Date): [DateFormat, TimeFormat] {
+        const newDate: DateFormat = { year: date.getFullYear(), month: date.getMonth() + 1, day: date.getDate() };
+        const newTime: TimeFormat = { hour: date.getHours(), minute: date.getMinutes() };
+        return [newDate, newTime];
+    }
+
+    /**
+     * Gets a DateFormat and a TimeFormat from a DateFormat and a TimeFormat Delta changed by x minutes.
+     * minutes can be postive or negative
+     */
```
Comment (domlimm) [inline src/web/services/datetime.service.ts:74]: Minor nit.

```suggestion
    /**
     * Gets a DateFormat and a TimeFormat from a DateFormat and a TimeFormat Delta changed by deltaMinutes.
     *
     * @param deltaMinutes accepts a postive or negative number.
     */
```

**Review by domlimm — COMMENTED** [review]
@gordonscottkolb Clean solution 👍🏻 Just a minor nit in the comment.

**Comment by kevin9foong** [comment]
Apart from @domlimm's comment, looks good to me. Thanks for your contribution!

**Review by domlimm — CHANGES_REQUESTED** [review]
@gordonscottkolb Please fix the lint error. Could refer to other `*.ts` files to see how the jsdocs is written. Also, please update your branch so we can merge it. Thanks!

File: src/web/services/datetime.service.ts [hunk 2]
```diff
@@ -52,6 +52,34 @@ export class DateTimeService {
...
+    convertDateToDateFormatAndTimeFormat(date: Date): [DateFormat, TimeFormat] {
+        const newDate: DateFormat = { year: date.getFullYear(), month: date.getMonth() + 1, day: date.getDate() };
+        const newTime: TimeFormat = { hour: date.getHours(), minute: date.getMinutes() };
+        return [newDate, newTime];
+    }
+
+    /**
+     * Gets a DateFormat and a TimeFormat from a DateFormat and a TimeFormat Delta changed by deltaMinutes.
+     *
+     * @param deltaMinutes accepts a postive or negative number.
```
Comment (kevin9foong) [inline src/web/services/datetime.service.ts:74]: Just need to add the description for the date and time here

File: src/web/services/datetime.service.ts [hunk 3]
```diff
@@ -52,6 +52,36 @@ export class DateTimeService {
...
 
+    /**
+     * Converts a DateFormat object and a TimeFormat object to a single Date instance.
+     */
+    convertDateFormatAndTimeFormatToDate(date: DateFormat, time: TimeFormat): Date {
+        return new Date(date.year, date.month - 1, date.day, time.hour, time.minute);
+    }
+
+    /**
+     * Converts a a single Date instance to a DateFormat object and a TimeFormat object.
```
Comment (domlimm) [inline src/web/services/datetime.service.ts:63]: Sorry just caught this.
```suggestion
     * Converts a single Date instance to a DateFormat object and a TimeFormat object.
```

File: src/web/services/datetime.service.ts [hunk 4]
```diff
@@ -52,6 +52,36 @@ export class DateTimeService {
...
+        const newDate: DateFormat = { year: date.getFullYear(), month: date.getMonth() + 1, day: date.getDate() };
+        const newTime: TimeFormat = { hour: date.getHours(), minute: date.getMinutes() };
+        return [newDate, newTime];
+    }
+
+    /**
+     * Gets a DateFormat and a TimeFormat from a DateFormat and a TimeFormat Delta changed by deltaMinutes.
+     *
+     * @param date accepts a DateFormat object.
+     * @param time accepts a TimeFormat object.
```
Comment (domlimm) [inline src/web/services/datetime.service.ts:75]:
```suggestion
     * @param date current date to change.
     * @param time current time to change.
```

**Review by domlimm — CHANGES_REQUESTED** [review]
@gordonscottkolb Last bit to update, and we should be good to merge! Thanks!

**Review by domlimm — APPROVED** [review]
@gordonscottkolb LGTM! Thank you for your contribution!

**Review by kevin9foong — APPROVED** [review]
LGTM!

**Comment by gordonscottkolb** [comment]
@kevin9foong @domlimm is there anything else I need to do for this issue, or will someone merge the pull request now?

**Comment by domlimm** [comment]
@gordonscottkolb That will be all. Thanks Gordon! 😃

**Pull request merged**